packages/code-block/utils/process-snippet.js

```diff
@@ -18,20 +18,51 @@ function processSnippet(snippet) {
  * @param {string} snippet
  */
 function parseShellSnippet(snippet) {
-  const firstLine = snippet.split('\n')[0]
-  // Is this shell snippet multiline?
-  const isMultiLine = firstLine.endsWith('\\') || firstLine.endsWith('EOF')
-  // If not multiline, return the single line snippet with leading '$ ' removed
-  if (!isMultiLine) return firstLine.replace('$ ', '')
-  // Else, return the multiline snippet formatted with shellwords escape & split
-  const multiLineFmt = shellwords.escape(snippet).replace('\\$\\', '')
-  // @TODO - re-assess use of shellwords.split() here. What are we trying to accomplish?
-  // It seems this splits commands into distinct tokens - but we want to copy all
-  // tokens of the command in their original format, don't we?
-  // ref: https://github.com/jimmycuadra/shellwords (really light on docs)
-  // ref: https://ruby-doc.org/stdlib-1.9.3/libdoc/shellwords/rdoc/Shellwords.html (ruby module of same name, has better docs)
-  // asana task: https://app.asana.com/0/1100423001970639/1199504357822173/f
-  return shellwords.split(multiLineFmt).join(' ')
+  const lines = snippet.split('\n')
+  /**
+   * Handle multi-line snippets.
+   *
+   * TODO: ideally we would detect many multi-line commands throughout
+   * a snippet. We could potentially build some complex string-based logic to
+   * do this... but it might be more efficient and effective to instead
+   * process the `code` passed to `HiddenCopyContent`:
+   * - We have incoming highlighted HTML or JSX, so we might be able to use
+   *   the highlight token classes (eg "shell-symbol") to only copy commands.
+   * - However, our highlighter currently doesn't distinguish between
+   *   "commands" and "output" tokens for the "shell-session" language...
+   *   So maybe complex string-based logic would be the way to go if
+   *   we want to support multiple multi-line commands in a single snippet?
+   * - Or maybe just need to look into `shellwords` use here?
+   *   (Full disclosure, I didn't set it up, so am not familiar.
+   *   it seems like it could be part of some solution?)
+   *
+   * ref: https://github.com/jimmycuadra/shellwords (really light on docs)
+   * ref: https://ruby-doc.org/stdlib-1.9.3/libdoc/shellwords/rdoc/Shellwords.html
+   *      (ruby module of same name, seems to have better docs)
+   *
+   * Related task:
+   * https://app.asana.com/0/1100423001970639/1199504357822173/f
```

Comment:
Would love to explore better copying behaviour at some point in the future... but for now, was definitely reaching for a more immediate solution to ensure our "copy these install commands" works on download pages. See hashicorp/dev-portal#1149.